docs: add SS 7.8 to 8.0 upgrade guide (#3522)
nordic-institute/X-Road: doc/Manuals/ig-cs_x-road_v8_central_server_upgrade_guide.md

| Date       | Version | Description     | Author             |
|------------|---------|-----------------|--------------------|
| 19.05.2026 | 1.0     | Initial version | Egidijus Milierius |

## License

This document is licensed under the Creative Commons Attribution-ShareAlike 3.0 Unported License. To view a copy of this license, visit http://creativecommons.org/licenses/by-sa/3.0/

## Table of Contents <!-- omit in toc -->

- [License](#license)
- [Overview](#overview)
  - [What this guide covers](#what-this-guide-covers)
  - [What this guide does not cover](#what-this-guide-does-not-cover)
  - [Disclaimer](#disclaimer)
- [Prerequisites](#prerequisites)
- [Before you start](#before-you-start)
- [Manual upgrade procedure](#manual-upgrade-procedure)
- [Post-upgrade verification](#post-upgrade-verification)
- [Rollback](#rollback)
- [Troubleshooting](#troubleshooting)
- [References](#references)

## Overview

This guide walks you through upgrading an existing X-Road 7.8 Central Server to X-Road 8 (beta 2). There is no Central Server upgrade wizard for beta 2; the procedure documented here drives `migration-cli.jar` and the package manager directly.

### What this guide covers

> - Single-node Central Server instances installed from native packages (DEB on Ubuntu, RPM on RHEL)

Reviewer: ```suggestion
- Single-node Central Server instances installed from native packages (DEB on Ubuntu)
```

Reviewer: Central Server doesn't support RHEL.

- Step-by-step manual procedure
- Post-upgrade verification
- Rollback options

### What this guide does not cover

- **Security Server upgrades** — see [Security Server Upgrade Guide](ig-ss_x-road_v8_security_server_upgrade_guide.md)
- **Central Server HA (cluster) upgrades** — see the [Central Server High Availability Installation Guide](ig-csha_x-road_6_ha_installation_guide.md) for the existing cluster topology; the X-Road 8 HA upgrade workflow is **not yet documented for beta 2**. A clustered Central Server requires shared-database coordination during upgrade — running this single-node procedure independently on every node would conflict with the replicated `centerui` database.

> - **Container or Kubernetes deployments** — covered in a separate document

Reviewer: Beta 2 doesn't support deploying Central Server using Docker so this should be removed.

- Upgrade from X-Road versions older than 7.8.x — must first be brought to 7.8.x

### Disclaimer

This document applies to X-Road 8 Beta 2. Pre-release software may behave differently from the final X-Road 8.0 release. **Central Server upgrades on beta 2 are operator-driven**; gaps in the procedure may exist and should be reported as GitHub issues.

## Prerequisites

| Requirement | Notes |
|---|---|

> | Operating system | Ubuntu Server 22.04 / 24.04 LTS, or RHEL 9 / 10 |

Reviewer: ```suggestion
| Operating system | Ubuntu Server 22.04 / 24.04 LTS |
```

```bash
PGPASSWORD='<centerui-password>' \
  psql -w -h <host> -p <port> -U <centerui-user> -d postgres \
  -tAc 'SHOW server_version_num'
```

The numeric server version divided by 10000 must be ≥ 15.

**5. Configure the OpenBao package repository.**

```bash
# Ubuntu
curl -fsSL https://openbao.org/assets/openbao-gpg-pub-20240618.asc \
  -o /usr/share/keyrings/openbao-keyring.asc
echo "deb [signed-by=/usr/share/keyrings/openbao-keyring.asc] https://pkgs.openbao.org/deb stable main" \
  > /etc/apt/sources.list.d/openbao.list
apt-get update

# RHEL
rpm --import https://openbao.org/assets/openbao-gpg-pub-20240618.asc
tee /etc/yum.repos.d/openbao.repo <<'EOF'
[openbao]
name=OpenBao
baseurl=https://pkgs.openbao.org/rpm
enabled=1
gpgcheck=1
gpgkey=https://openbao.org/assets/openbao-gpg-pub-20240618.asc
EOF
dnf makecache
```

**6. Stop all Central Server services.**

```bash
mapfile -t SERVICES < <(systemctl list-units --type=service --state=active 'xroad-*' \
  --no-legend --plain | awk '{print $1}')
for s in "${SERVICES[@]}"; do
  systemctl stop "$s"
  for _ in {1..30}; do systemctl is-active --quiet "$s" || break; sleep 2; done
done
```

> The Central Server service set typically includes `xroad-center`, `xroad-center-management-service`, `xroad-center-registration-service`, `xroad-signer`, `xroad-monitor`, and optionally `xroad-opmonitor`.

Reviewer: ```suggestion
The Central Server service set typically includes `xroad-center`, `xroad-center-management-service`, `xroad-center-registration-service`, `xroad-signer` and `xroad-nginx`.
```
